xds/src/main/java/io/grpc/xds/XdsServerBuilder.java

```diff
@@ -138,4 +152,100 @@ public void onNotServing(Throwable throwable) {
       notServing = true;
     }
   }
+
+  @Override
+  public XdsServerBuilder directExecutor() {
+    checkState(!isServerBuilt.get(), "Server already built!");
```

Comment:
It seems the `delegate` builder must not change its state through the whole lifecycle of parent server. But there is nothing can be done here on the delegate builder to prevent it being changed.

Reply:
The check is not for build() but for the setters in XdsServerBuilder(). This extends `ForwardingServerBuilder` and that one just calls the method on the delegate e.g. `ForwardingServerBuilder#maxInboundMetadataSize`
```
  public T maxInboundMetadataSize(int bytes) {
    delegate().maxInboundMetadataSize(bytes);
    return thisT();
  }
``` 
But the `XdsServerBuilder` build() produces `ServerWrapperForXds` which can reuse the delegate builder later (because of listener delete/add semantics) we want to prevent the delegate builder from getting modified because of the outer `XdsServerBuilder`.